custom_components/dualmode_generic/climate.py

```diff
@@ -470,12 +526,26 @@ async def _async_control_heating(self, time=None, force=False):
                         await self._async_cooler_turn_off()
                     elif self._hvac_mode == HVAC_MODE_HEAT:
                         await self._async_heater_turn_off()
+                    elif self._hvac_mode == HVAC_MODE_FAN_ONLY:
+                        await self._async_fan_turn_off()
+
+            if self.fan_behavior == FAN_MODE_NEUTRAL and self._hvac_mode == HVAC_MODE_FAN_ONLY:
+                if self.fan_entity_id is not None:
+                    await self._async_fan_turn_on()
 
     @property
     def _is_device_active(self):
         """If the toggleable device is currently active."""
-        return self.hass.states.is_state(self.heater_entity_id, STATE_ON) or \
-               self.hass.states.is_state(self.cooler_entity_id, STATE_ON)
+        device_state_list = [self.hass.states.is_state(self.heater_entity_id, STATE_ON),
+                             self.hass.states.is_state(self.cooler_entity_id, STATE_ON)]
+
+        if self.fan_entity_id is not None:
+            device_state_list.append(self.hass.states.is_state(self.fan_entity_id, STATE_ON))
+
+        for state in device_state_list:
```

Comment:
This whole function could be simplified to:
```
def _is_device_active(self):
  devices = [self.heater_entity_id, self.cooler_entity_id] + ([self.fan_entity_id] if self.fan_entity_id else [])
  device_states = [self.hass.states.is_state(dev, STATE_ON) for dev in devices]
  return next((state for state in device_states if state), False)
```

Reply:
I'm grateful for feedback. I just started coding with Python last week ;D